falcon/statics.py

```diff
@@ -0,0 +1,43 @@
+import copy
+import mimetypes
+import os.path
+
+import falcon
+
+
+class StaticSink:
+
+    _additional_types = {
+        '.css': 'text/css',
+        '.js': 'application/javascript',
+        '.html': 'text/html',
+        '.svg': 'image/svg+xml',
+
+        '.png': 'image/png',
+        '.jpeg': 'image/jpeg',
+        '.jp2': 'image/jpeg',
+        '.gif': 'image/gif',
+        '.tiff': 'image/tiff',
+    }
+
+    def __init__(self, prefix, parent_folder):
+        self.prefix = prefix
+        self.parent_folder = os.path.abspath(parent_folder)
+        self.mime_types = copy.copy(mimetypes.encodings_map)
+
+        self.mime_types.update(self._additional_types)
+
+    def __call__(self, req, resp):
+        stripped = req.path.replace(self.prefix, '')
+        cleaned = stripped.replace('../', '').lstrip('/')
```

Comment:
It may be better to instead run `stripped` through `os.path.normpath()` and then explicitly raise `falcon.HTTPNotFound()` if the result starts with `'..'` or `/` (rather than checking first if the file exists with those characters stripped). This has the added benefits of allowing harmless redundant separators and level references, replacing forward slashes with backslashes on Windows, and ensuring that malformed requests aren't inadvertently mapped to existing files.